project_timesheet_management/models/timesheet_invoice.py

```diff
@@ -0,0 +1,764 @@
+                    email_values={"author_id": author_id or False})
+
+            # Supplier Project Coordinator
+            user_ids = timesheet_inv.project_id.favorite_user_ids.filtered(
+                lambda user: user.has_group(po_group) and
+                user.id not in u_lst and (
+                    user.partner_id == timesheet_inv.partner_id or
+                    user.supplier_company_id == timesheet_inv.partner_id)) or \
+                timesheet_inv.project_id.resource_ids.mapped(
+                    'user_ids').filtered(
+                        lambda user: user.has_group(po_group) and
+                user.id not in u_lst and (
+                    user.partner_id == timesheet_inv.partner_id or
+                    user.supplier_company_id == timesheet_inv.partner_id))
+    
+            for user_id in user_ids:
+                mail_template.write({
+                    'email_to': user_id.email or '',
+                    'email_from': self.env.user.email or self.env.user.partner_id.email,
+                    'reply_to': self.env.user.email or self.env.user.partner_id.email,})
+                mail_template.sudo().with_context(state=mail_status).send_mail(
+                    timesheet_inv.id, force_send=True,
+                    email_values={'author_id': author_id or False})
+
+            # Canna Project Coordinator
+            u_lst += user_ids.ids
+            c_user_ids = timesheet_inv.project_id.favorite_user_ids.filtered(
+                lambda user: user.has_group(
+                    cann_po_group) and user.id not in u_lst) or \
+                timesheet_inv.project_id.resource_ids.mapped(
+                    'user_ids').filtered(
+                    lambda user: user.has_group(
+                        cann_po_group) and user.id not in u_lst)
+            for user_id in c_user_ids:
+                mail_template.write({
+                    'email_to': user_id.email or '',
+                    'email_from': self.env.user.email or self.env.user.partner_id.email,
+                    'reply_to': self.env.user.email or self.env.user.partner_id.email,})
+                mail_template.sudo().with_context(state=mail_status).send_mail(
+                    timesheet_inv.id, force_send=True,
+                    email_values={'author_id': author_id or False})
+
+    def unlink(self):
+        """
+        @override
+        Preventing record deletion if the record is approved.
+        Raises:
+            UserError: If the record is Approved
+        """
+        for record in self:
+            if record.state == "approved":
+                raise UserError(
+                    _(
+                        "You can not delete Timesheet Invoice "
+                        "which is already Approved."
+                    )
+                )
+        return super(TimesheetInvoice, self).unlink()
+
+    @api.returns("self", lambda value: value.id)
+    def message_post(self, **kwargs):
```

Comment:
This can be removed if you are not modifying it

Reply:
Keeping for now, will verify in old version and update later